packages/config/config/devices/0x024e/WIN-GDC-02.json

```diff
@@ -0,0 +1,323 @@
+{
+	"manufacturer": "zConnect",
+	"manufacturerId": "0x024e",
+	"label": "DHS-WIN-GDC-02",
+	"description": "Garage Door Opener",
+	"devices": [
+		{
+			"productType": "0x0021",
+			"productId": "0x0014"
+		}
+	],
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"endpoints": {
```

Comment:
So have plugged this config manually into zwavejs2mqtt in home assistant, and the endpoint labels aren't coming up. Have I done something wrong here? I tried just dropping it back to associations only, but that didn't work either.

Reply:
Hard to tell without the device. Maybe a node dump from z2m could help diagnose.

Reply:
Seems to have multiple copies of the same information:

```
    "groups": [
      {
        "text": "Unnamed group 1",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 2",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 3",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 4",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 5",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 1",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 2",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 1",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 2",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 3",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 4",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 5",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 1",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 2",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 1",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 2",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 3",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 4",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 5",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "BS1 via Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "On/Off (BS1)",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "BS2 via Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "On/Off (BS2)",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 1",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 2",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 3",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 4",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 5",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 1",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 2",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 1",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Sensor 2",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Sensor 3",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Sensor 4 (optional temp sensor)",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "BS1 via Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "On/Off (BS1)",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "BS2 via Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "On/Off (BS2)",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 1",
        "endpoint": 0,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 2",
        "endpoint": 0,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 3",
        "endpoint": 0,
        "value": 3,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 4",
        "endpoint": 0,
        "value": 4,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Unnamed group 5",
        "endpoint": 0,
        "value": 5,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 1,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 1",
        "endpoint": 1,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      },
      {
        "text": "Lifeline",
        "endpoint": 2,
        "value": 1,
        "maxNodes": 3,
        "isLifeline": true,
        "multiChannel": true
      },
      {
        "text": "Sensor 2",
        "endpoint": 2,
        "value": 2,
        "maxNodes": 10,
        "isLifeline": false,
        "multiChannel": true
      }
    ],
```

Reply:
Weird... @robertsLando any idea why there are so many copies of some groups?

Reply:
@Swamp-Ig just to be sure - did you re-interview the device after adding the config?

Reply:
Yes, several times. Plus restarted zwavejs2mqtt several times too.

Reply:
Hmm I just do this: https://github.com/zwave-js/zwavejs2mqtt/blob/master/lib/ZwaveClient.ts#L717

Reply:
So... you don't clear the `groups` array when another call to `getGroups` is made?

Reply:
https://github.com/zwave-js/zwavejs2mqtt/pull/1391

Reply:
So it would appear my work is good to go then. I can see the right data is in there, just mixed up with the wrong data.

Reply:
.. err... so I tried manually clearing the groups list:

```
driver.controller.nodes.get(6).groups = [];
```

and then re-interviewing again. I get the right number of groups, but the labels for group 0 are still wrong. The weird thing is that the labels for endpoint 1 and 2 have actually copied across from endpoint 0 correctly. I suspect the above PR has only got part way.

Unfortunately I don't know the code well enough to fix the bug myself.


```
  "groups": [
    {
      "text": "Unnamed group 1",
      "endpoint": 0,
      "value": 1,
      "maxNodes": 3,
      "isLifeline": true,
      "multiChannel": true
    },
    {
      "text": "Unnamed group 2",
      "endpoint": 0,
      "value": 2,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    },
    {
      "text": "Unnamed group 3",
      "endpoint": 0,
      "value": 3,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    },
    {
      "text": "Unnamed group 4",
      "endpoint": 0,
      "value": 4,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    },
    {
      "text": "Unnamed group 5",
      "endpoint": 0,
      "value": 5,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    },
    {
      "text": "Lifeline-moo",
      "endpoint": 1,
      "value": 1,
      "maxNodes": 3,
      "isLifeline": true,
      "multiChannel": true
    },
    {
      "text": "Sensor 1",
      "endpoint": 1,
      "value": 2,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    },
    {
      "text": "Lifeline-moo",
      "endpoint": 2,
      "value": 1,
      "maxNodes": 3,
      "isLifeline": true,
      "multiChannel": true
    },
    {
      "text": "Sensor 2",
      "endpoint": 2,
      "value": 2,
      "maxNodes": 10,
      "isLifeline": false,
      "multiChannel": true
    }
  ],
```

Reply:
I'll see if I can reproduce this in a test case later.

Reply:
We've just gone into COVID lockdown so I might even have a bit of time to have a go with it 😁 

It looks to me like what I've done is according to spec, you could potentially merge it if you wanted and fix the underlying code later, but that's your call of course.

Reply:
Confirmed bug. I'll merge this and fix then.